core/src/main/scala/org/apache/spark/deploy/master/Master.scala

```diff
@@ -467,7 +467,9 @@ private[spark] class Master(
    * two executors on the same worker).
    */
   def canUse(app: ApplicationInfo, worker: WorkerInfo): Boolean = {
-    worker.memoryFree >= app.desc.memoryPerSlave && !worker.hasExecutor(app)
+    worker.memoryFree >= app.desc.memoryPerSlave &&
+    !worker.hasExecutor(app) &&
+    !app.removedExecutors.exists(_.worker == worker)
```

Comment:
@ksakellis thanks for your comment. Blacklist with TTL is a good idea, but I think it's a little complicated in this scenario. Furthermore, `removedExecutors` is a member variable in `ApplicationInfo`, so it can only take effect in the current application, not the whole cluster.